t/unit/tasks/test_canvas.py

```diff
@@ -696,6 +710,28 @@ def test_group_stamping_two_levels(self):
         assert second_nested_sig_res._get_task_meta()['groups'] == \
                [g1_res.id, g2_res.id]
 
+    @pytest.mark.usefixtures('depends_on_current_app')
+    def test_group_stamping_with_replace(self):
+        """
+        For a group within a group, test that group stamps are stored in
+        the correct order.
+        """
+        self.app.conf.task_always_eager = True
+        self.app.conf.task_store_eager_result = True
+        self.app.conf.result_extended = True
+
+        sig_1 = self.add.s(2, 2)
+        sig_2 = self.add.s(2, 2) | self.replaced.s(8)
+        sig_1_res = sig_1.freeze()
+        sig_2_res = sig_2.freeze()
+
+        g = group(sig_1, sig_2, app=self.app)
+        g_res = g.freeze()
+        g.apply()
+
+        assert sig_1_res._get_task_meta()['groups'] == [g_res.id]
+        assert sig_2_res._get_task_meta()['groups'] == [g_res.id]
```

Comment:
Are we sure this checks the result of `replaced`?

Reply:
sig_2_res should be stamped with g_res.id
sig_2_res would without stamping have an empty groups list, as this is result of replaced application